api/s2n.h

```diff
@@ -54,6 +54,7 @@ extern int s2n_config_free_dhparams(struct s2n_config *config);
 extern int s2n_config_free_cert_chain_and_key(struct s2n_config *config);
 extern int s2n_config_set_nanoseconds_since_epoch_callback(struct s2n_config *config, int (*nanoseconds_since_epoch)(void *, uint64_t *), void * data);
 extern const char *s2n_strerror(int error, const char *lang);
+extern const char *s2n_strerror_debug(void);
```

Comment:
I think that would be ideal! There are a couple blockers that may require redesigning our error strings:
* There's no mapping between an s2n_errno and the s2n_debug_str. If i give `s2n_strerror(some_error, "EN", S2N_DEBUG)` how do we know to which debug string to give back (other than the most recent one)? And of course we can't assume the most recent error is what the caller was interested in.
* Returning a dynamic allocated string. If we want to return some concatenation of the s2n error string and the debug string to the caller it'd need to be dynamically allocated. Not sure if we want to go through the trouble of exposing that in the api.

The current change is not without its problems. Something like this would be incorrect:
```
if (my_s2n_errno = some_recorded_s2n_errno()) {
    record_tls_error(s2n_strerror(my_s2n_errno, "EN"));
    /* Gives most recent debug string, may not be associated with my_s2n_errno */
    record_tls_error_debug(s2n_strerror_debug());
}
```